stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/view/FootnoteView.kt

```diff
@@ -0,0 +1,115 @@
+package io.getstream.chat.android.ui.messages.adapter.view
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import android.widget.TextView
+import androidx.annotation.DrawableRes
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.constraintlayout.widget.ConstraintSet
+import androidx.core.view.isVisible
+import com.getstream.sdk.chat.utils.extensions.constrainViewToParentBySide
+import com.getstream.sdk.chat.utils.extensions.inflater
+import com.getstream.sdk.chat.utils.extensions.updateConstraints
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.databinding.StreamUiItemMessageFootnoteBinding
+import io.getstream.chat.android.ui.databinding.StreamUiMessageThreadsFootnoteBinding
+
+internal class FootnoteView : ConstraintLayout {
+
+    constructor(context: Context) : super(context)
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) : super(
+        context,
+        attrs,
+        defStyleAttr,
+        defStyleRes
+    )
+
+    private val footnote: StreamUiItemMessageFootnoteBinding =
+        StreamUiItemMessageFootnoteBinding.inflate(inflater).also { addView(it.root) }
```

Comment:
Don't we normally use `apply` in those cases?

Reply:
I don't remember any strong obligations about such use-case. I always see this `also` extension as side-effect. Here we initialize by inflating the binding, and add its root to the view.